### ISSUE TITLE
date-fns module

### ISSUE THREAD

**Issue by mcgri** [mattlewis92/angular2-calendar#74]
Hello
I am trying to integrate your  calendar into my project, so I started with a demo..
But immediately I ran into a problem with date-fns dependancy. The typescript just can't find it.

I am using systemjs. 
Here is my package.json

`
    "@angular/common": "~2.0.1",
    "@angular/compiler": "~2.0.1",
    "@angular/core": "~2.0.1",
    "@angular/forms": "~2.0.1",
    "@angular/http": "~2.0.1",
    "@angular/platform-browser": "~2.0.1",
    "@angular/platform-browser-dynamic": "~2.0.1",
    "@angular/router": "~3.0.1",
    "@angular/upgrade": "~2.0.1",
    "angular-in-memory-web-api": "~0.1.1",
    "bootstrap": "^3.3.7",
    "core-js": "^2.4.1",
    "reflect-metadata": "^0.1.8",
    "rxjs": "5.0.0-beta.12",
    "systemjs": "0.19.39",
    "zone.js": "^0.6.25",
   "chart.js": "1.1.1",

    "ng2-ckeditor": "^1.0.3",
    "ng2-file-upload": "^1.0.3",
    "ng2-img-cropper": "^0.6.4",
    "angular2-infinite-scroll": "^0.2.4",
    "angular2-calendar": "^0.3.5",
    "calendar-utils": "^0.0.30",
    "date-fns": "^1.3.0",
    "@types/date-fns": "^0.0.2"`

**Comment by mattlewis92** [comment]
What do your imports from the date-fns lib looks like?

**Comment by mcgri** [comment]
Here it is
`
import {
    startOfDay,
    subDays,
    addDays,
    endOfMonth,
    isSameDay,
    isSameMonth,
    addWeeks,
    subWeeks,
    addMonths,
    subMonths
} from 'date-fns';


import {
    CalendarEvent,
    CalendarEventAction
} from 'angular2-calendar';`

**Comment by mattlewis92** [comment]
What error are you getting from typescript? Are you using typescript 2.0.0+?

**Comment by mcgri** [comment]
the exact error is "Error		Build:Cannot find module 'date-fns'."  I am using TS ^2 . 
Just checked in console - $ tns --version 2.3.0 . I will check the project settings...maybe something went wrong with TS update

**Comment by mattlewis92** [comment]
I think the issue is with System.js as you need to configure it to map the import to the date-fns lib in your system config, I'm no expert on system.js though so I'd recommend making a minimal repro and posting it on stackoverflow where you'd be able to get a better level of support.

**Comment by mcgri** [comment]
Yeah, played with system.js config for hours. Okay, thanks, Matt. I will post my solution here when I find one

**Comment by mattlewis92** [comment]
If you can I'd highly recommend going with the [angular CLI](https://github.com/angular/angular-cli) for building your app, there's far less config and moving parts than with system.js 😄

**Comment by mcgri** [comment]
Haha =)  I already experienced a development of 60+ components with angular beta 17 -> Final release :))
I don't have enough nerves to do it again with CLI. I might consider it after it will become final :))

Anyway, your questions actually helped to resolve my problem. Turned out what I didn't update TS version in the project
`<TypeScriptToolsVersion>2.0.3</TypeScriptToolsVersion> `

Plus after some woodo with systemjs , It finally works...sort of. I will be to figure out where is the css for your demo :)

Oh. I also had to include few extra providers:

`import { CalendarModule as ng2Calendar, CalendarDateFormatter, CalendarEventTitle} from 'angular2-calendar';
import { CalendarComponent } from "../../ngComponents/ngComponents";

@NgModule({
    imports: [BrowserModule, ng2Calendar ],
    declarations: [CalendarComponent],
    providers: [CalendarDateFormatter, CalendarEventTitle],`

Anyway. Thanks for such quick replies!
